packages/build-tools/src/steps/functions/download_build.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  BuildFunction,
+  BuildStepInput,
+  BuildStepInputValueTypeName,
+  BuildStepOutput,
+} from '@expo/steps';
+
+export function createDownloadBuildFunction(): BuildFunction {
+  return new BuildFunction({
+    namespace: 'eas',
+    id: 'download_build',
+    name: 'Download build',
+    inputProviders: [
+      BuildStepInput.createProvider({
+        id: 'build_id',
+        required: true,
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+      }),
+    ],
+    outputProviders: [
+      BuildStepOutput.createProvider({
+        id: 'artifact_path',
+        required: false,
+      }),
+    ],
+    command: `
```

Comment:
It doesn't need to be done for v1, let's iterate on it later on, but I would personally suggest having `eas/download_build` as TS function vs bash script

I believe using TS, libraries and our other existing utils would make it less error-prone